proxy-node-gateway/src/main/java/uk/gov/ida/notification/services/EidasSamlParserService.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.ida.notification.services;
+
+import uk.gov.ida.jerseyclient.JsonClient;
+import uk.gov.ida.notification.dto.EidasSamlParserRequest;
+import uk.gov.ida.notification.dto.EidasSamlParserResponse;
+import uk.gov.ida.notification.exceptions.EidasSamlParserResponseException;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import java.net.URI;
+import java.util.Set;
+
+public class EidasSamlParserService {
+    private final JsonClient eidasSamlParserClient;
+    private final URI eidasSamlParserURI;
+    private final Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
+
+    public EidasSamlParserService(
+            JsonClient eidasSamlParserClient,
+            URI eidasSamlParserURI) {
+        this.eidasSamlParserClient = eidasSamlParserClient;
+        this.eidasSamlParserURI = eidasSamlParserURI;
+    }
+
+    public EidasSamlParserResponse parse(EidasSamlParserRequest eidasSamlParserRequest) {
+        EidasSamlParserResponse response = eidasSamlParserClient.post(eidasSamlParserRequest, eidasSamlParserURI, EidasSamlParserResponse.class);
+
+        return validateEidasSamlResponseParserDTO(response);
+    }
+
+    private EidasSamlParserResponse validateEidasSamlResponseParserDTO(EidasSamlParserResponse eidasSamlParserResponse) {
+        Set<ConstraintViolation<EidasSamlParserResponse>> violations = validator.validate(eidasSamlParserResponse);
```

Comment:
feeling a bit uneasy about having to validate the @NotBlank annotations "by hand". Is it possible to use `javax.validation.constraints.Min(1)` or `javax.validation.constraints.Pattern(regex)` instead

Reply:
Min is used for numbers by the looks of it, but we could used a regex. I'm not sure what the benefit would be though? All it's currently validating is that the values aren't empty strings or null. How do you mean "by hand"? It'd be the same process - annotating the field then calling `validate` on the object?

Reply:
You're right, Min is for numbers 🤭. I had thought using an annotation in package `javax.validation.constraints` meant the annotation and field would be evaluated at marshalling time by jersey, rather than us having to create validation code.

Reply:
Yeah, me too! It doesn't seem to though. If I change the annotations, the validation call will still need to happen I think.

Reply:
Pretty sure Dropwizard takes care of `@NotNull` and `@NotBlank` validations provided you supply `@Valid` as an annotation to the method parameter.

Reply:
Although maybe that's only for Resources :thinking:

Reply:
Yeah, Phil had a look at this, it looks like `@Valid` works on the server side (server returns a 500 if the object isn't valid) but it doesn't work for validating client side. As this is from a trusted thing, validating client side is probably unnecessary so I'm just going to remove it.